features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/base64/ImageWireframeHelper.kt

```diff
@@ -65,7 +76,179 @@ internal class ImageWireframeHelper(
         return imageWireframe
     }
 
-    private companion object {
-        private const val DRAWABLE_CHILD_NAME = "drawable"
+    @Suppress("NestedBlockDepth")
+    internal fun createCompoundDrawableWireframes(
+        view: TextView,
+        mappingContext: MappingContext,
+        startIndex: Int
+    ): MutableList<MobileSegment.Wireframe> {
+        val result = mutableListOf<MobileSegment.Wireframe>()
+        var currentIndex = startIndex
+
+        view.compoundDrawables.indices.forEach {
+            val drawable = view.compoundDrawables[it]
+            val density = mappingContext.systemInformation.screenDensity
+
+            if (drawable != null) {
+                val compoundDrawableLocationMap = createCompoundDrawableLocationMap(
+                    view = view,
+                    drawable = drawable,
+                    density = density,
+                    index = it
+                )
+
+                val drawableX = compoundDrawableLocationMap[it]?.first
+                val drawableY = compoundDrawableLocationMap[it]?.second
+
+                if (drawableX != null && drawableY != null) {
+                    @Suppress("ThreadSafety") // TODO REPLAY-1861 caller thread of .map is unknown?
+                    createImageWireframe(
+                        view = view,
+                        index = ++currentIndex,
+                        x = drawableX,
+                        y = drawableY,
+                        width = drawable.intrinsicWidth
+                            .densityNormalized(density).toLong(),
+                        height = drawable.intrinsicHeight
+                            .densityNormalized(density).toLong(),
+                        drawable = drawable,
+                        shapeStyle = null,
+                        border = null
+                    )?.let { resultWireframe ->
+                        result.add(resultWireframe)
+                    }
+                }
+            }
+        }
+
+        return result
+    }
+
+    private fun createCompoundDrawableLocationMap(
+        view: TextView,
+        drawable: Drawable,
+        density: Float,
+        index: Int
+    ): HashMap<Int, Pair<Long, Long>> {
```

Comment:
return type can be just `Map`

```suggestion
    ): Map<Int, Pair<Long, Long>> {
```